aiida/storage/sqlite_zip/migrations/legacy_to_main.py

```diff
@@ -98,7 +98,28 @@ def _in_archive_context(_inpath):
             temp_folder = working / 'temp_unpack'
             with tarfile.open(_inpath, 'r') as tar:
                 MIGRATE_LOGGER.report('Extracting tar archive...(may take a while)')
-                tar.extractall(temp_folder)
+
+                import os
+
+                def is_within_directory(directory, target):
+
+                    abs_directory = os.path.abspath(directory)
+                    abs_target = os.path.abspath(target)
+
+                    prefix = os.path.commonprefix([abs_directory, abs_target])
+
+                    return prefix == abs_directory
+
+                def safe_extract(tar, path='.', members=None, *, numeric_owner=False):
+
+                    for member in tar.getmembers():
+                        member_path = os.path.join(path, member.name)
+                        if not is_within_directory(path, member_path):
+                            raise Exception('Attempted Path Traversal in Tar File')
+
+                    tar.extractall(path, members, numeric_owner)
```

Comment:
According to the [official docs](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extractall) `numeric_owner` is a keyword-only argument.
```suggestion
                    tar.extractall(path, members, numeric_owner=numeric_owner)
```